src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryItem.cs

```diff
@@ -246,9 +246,10 @@ internal bool EqualsBasemap(Basemap? other)
                 return false;
             }
 
-            return other == Basemap || other.Name == Basemap?.Name
-                                    || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
-                                    || (other.Uri != null && other.Uri == Basemap?.Uri);
+            return other == Basemap || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
+                                    || (other.Uri != null && other.Uri == Basemap?.Uri)
+                                    || (Basemap?.BaseLayers.Count == other.BaseLayers.Count
+                                        && Basemap.BaseLayers.All(layer => other?.BaseLayers.FirstOrDefault(l => l.Name == layer.Name) is not null));
```

Comment:
Still don't rely on names - rely on the actual content of the BaseLayers and also just as importantly ReferenceLayers. You'll need to compare each layer by type and Source uri.

Reply:
```suggestion
                                        && Basemap.BaseLayers.Zip(other.BaseLayers, (a, b) => new { LayerA = a, LayerB = b })
                                               .All(pair => pair.LayerA.GetType() == pair.LayerB.GetType()
                                                     && (pair.LayerA as ArcGISSceneLayer)?.Source == (pair.LayerB as ArcGISSceneLayer)?.Source
                                                     && (pair.LayerA as ArcGISTiledLayer)?.Source == (pair.LayerB as ArcGISTiledLayer)?.Source
                                                     && (pair.LayerA as ArcGISVectorTiledLayer)?.Source == (pair.LayerB as ArcGISVectorTiledLayer)?.Source)
                                        && Basemap.ReferenceLayers.Zip(other.ReferenceLayers, (a, b) => new { LayerA = a, LayerB = b })
                                               .All(pair => pair.LayerA.GetType() == pair.LayerB.GetType()
                                                     && (pair.LayerA as ArcGISSceneLayer)?.Source == (pair.LayerB as ArcGISSceneLayer)?.Source
                                                     && (pair.LayerA as ArcGISTiledLayer)?.Source == (pair.LayerB as ArcGISTiledLayer)?.Source
                                                     && (pair.LayerA as ArcGISVectorTiledLayer)?.Source == (pair.LayerB as ArcGISVectorTiledLayer)?.Source));
```

How about this?

Reply:
Reflection isn't an option since that won't work in some release builds.
It might be cleaner to make an layer-comparer method you can use to simplify the overall expression